storage/client/src/main/java/com/azure/storage/blob/AppendBlobClientBuilder.java

```diff
@@ -60,20 +73,30 @@ public AppendBlobClientBuilder() {
         policies = new ArrayList<>();
     }
 
-    /**
-     * Constructs an instance of AppendBlobAsyncClient based on the configurations stored in the appendBlobClientBuilder.
-     * @return a new client instance
-     */
-    private AzureBlobStorageImpl buildImpl() {
+    private AzureBlobStorageBuilder buildImpl() {
         Objects.requireNonNull(endpoint);
+        Objects.requireNonNull(containerName);
+        Objects.requireNonNull(blobName);
 
         // Closest to API goes first, closest to wire goes last.
         final List<HttpPipelinePolicy> policies = new ArrayList<>();
 
+        if (configuration == null) {
+            configuration = ConfigurationManager.getConfiguration();
+        }
         policies.add(new UserAgentPolicy(BlobConfiguration.NAME, BlobConfiguration.VERSION, configuration));
         policies.add(new RequestIdPolicy());
         policies.add(new AddDatePolicy());
-        policies.add(credentials); // This needs to be a different credential type.
+
+        if (sharedKeyCredential != null) {
+            policies.add(new SharedKeyCredentialPolicy(sharedKeyCredential));
+        } else if (tokenCredential != null) {
```

Comment:
This is essentially saying if multiple credentials are present, we just pick one. Perhaps we should throw?